src/coap_blockwise.h

```diff
@@ -27,8 +27,9 @@ enum golioth_status golioth_blockwise_post(struct golioth_client *client,
 
 /* Blockwise Download */
 typedef enum golioth_status (*write_block_cb)(uint32_t block_idx,
-                                              uint8_t *block_buffer,
                                               size_t block_size,
+                                              uint8_t *block_buffer,
+                                              size_t current_block_len,
                                               bool is_last,
                                               void *callback_arg);
```

Comment:
I think we should rearrange/rename some of these:
- I think it makes sense to keep all the fields relating to the current block together (`block_idx`, `block_buffer`, `current_block_len`, and `is_last`). So I would move `block_size` to just before `callback_arg`.
- It will be clearer if we keep the buffer and length naming consistent - so instead of `current_block_len`, we could use `block_buffer_len`.
- Finally, to further remove ambiguity, renaming `block_size` to `negotiated_block_size` should make things crystal clear.